frontend/src/index.jsx

```diff
@@ -37,3 +38,22 @@ ReactDOM.render(
   </Provider>,
   document.getElementById('root')
 );
+
+// If there is a view generated in case one doesn't exist that view and the params are added to the URI and recorded to the history
+// as a consequence when the user hits the Back button has the impression that it is on the same page as the browser is visiting the same view
+// to deal with this case we added a `popstate` listener that will go to the correct page in history skipping
+// the case when the URL and the view are the same when the back button is pressed
+window.addEventListener('popstate', () => {
+  const appState = store.getState().appHandler;
+  const { lastBackPage } = appState;
+
+  if (
+    lastBackPage &&
+    lastBackPage.includes('viewId') &&
+    lastBackPage === document.location.href
+  ) {
+    window.history.back(-2);
```

Comment:
why not try to make sure we are not adding 2 entries to the history?

Reply:
The app was thought to work like that and push to history for the case when there is no view. If I would not push/replace the entries to history it will break the app, things will not load, filters will not work, ws connectors will not work as expected since they will not subscribe to the appropriate view aso. Overall I will be going into a rabbit hole and the budget allocated for this issue will be way exceeded. This is the reason why I chosen this solution.  Pls advice what to do. Thanks

Reply:
at least you could extract that this function and put it into that utils/index.js file, near that updateUri file which is manipulating the history in this way that you have t write this weird code here too.

Also, I would document why we have to go back(-2)... i.e. make a reference to that updateUri function.

In this way, the developers which will come after you and maybe even you from the future will be thankful.

Reply:
updated. Thanks